src/os_crypto/shared/msgs.c

```diff
@@ -407,10 +426,11 @@ int ReadSecMSG(keystore *keys, char *buffer, char *cleartext, int id, unsigned i
                 msg_local,
                 keys->keyentries[id]->global,
                 keys->keyentries[id]->local);
-
+            w_mutex_unlock(&keys->keyentries[id]->mutex);
             merror(ENCTIME_ERROR, keys->keyentries[id]->name);
             return KS_RIDS;
         }
+        w_mutex_unlock(&keys->keyentries[id]->mutex);
```

Comment:
This is not necessary, you are already unlocking the mutex at the end of the function.